packages/dag4-wallet/src/shared/operations.ts

```
@@ -0,0 +1,185 @@
+    allowSpendResponse = await network.l1Api.postAllowSpend(signedAllowSpend);
+  } catch (err) {
+    throw new Error("Error sending the allow spend transaction");
+  }
+
+  if (!allowSpendResponse || !allowSpendResponse.hash) {
+    throw new Error("Unable to get allow spend response");
+  }
+
+  return allowSpendResponse;
+};
+
+const tokenLock = async (
+  body: TokenLockWithCurrencyId,
+  network: SharedNetwork,
+  keyTrio: KeyTrio
+): Promise<HashResponse> => {
+  validateSchema(body, tokenLockSchema, true);
+
+  if (body.source !== keyTrio.address) {
+    throw new Error('"source" must be the same as the account address');
+  }
+
+  let tokenLockLastRef: TransactionReference | null = null;
+  let signedTokenLock: SignedTokenLock | null = null;
+  let tokenLockResponse: HashResponse | null = null;
+
+  try {
+    // Get token lock last reference
+    tokenLockLastRef = await network.l1Api.getTokenLockLastRef(keyTrio.address);
+  } catch (err) {
+    throw new Error("Error getting the token lock last reference");
+  }
+
+  if (!tokenLockLastRef) {
+    throw new Error("Unable to find token lock last reference");
+  }
+
+  try {
+    // Generate signed token lock body
+    const tokenLockBody: TokenLockWithParent = {
+      ...body,
+      parent: tokenLockLastRef,
+      currencyId: body.currencyId ?? null,
+      fee: body.fee ?? 0,
+      unlockEpoch: body.unlockEpoch ?? null,
+    };
+    signedTokenLock = await keyStore.generateBrotliSignature(
+      tokenLockBody,
+      normalizePublicKey(keyTrio.publicKey),
+      keyTrio.privateKey
+    );
+  } catch (err) {
+    throw new Error("Error generating the signed token lock");
+  }
+
+  if (!signedTokenLock) {
+    throw new Error("Unable to generate signed token lock");
+  }
+
+  try {
+    // Post signed token lock body
+    tokenLockResponse = await network.l1Api.postTokenLock(signedTokenLock);
+  } catch (err) {
+    throw new Error("Error sending the token lock transaction");
```

Same here - swallows the actual error.